api/src/utils/get-permissions.ts

```diff
@@ -24,7 +24,35 @@ export async function getPermissions(accountability: Accountability, schema: Sch
 		const cachedPermissions = await systemCache.get(cacheKey);
 
 		if (cachedPermissions) {
-			return cachedPermissions;
+			if (cachedPermissions.containDynamicData) {
```

Comment:
Maybe we can do early returns like

```js
if (!cachedPermissions.containDynamicData) {
    return processPermissions(accountability, cachedPermissions.permissions, {});
}
```

and the other if elses when applicable to reduce some nestings here. Just a thought, not really important 😄

Reply:
Agreed! Will refactor it soon! Thanks for the tip~ 😁